### DISCUSSION TITLE
Remove magic `import mill._` in build script

### DISCUSSION THREAD

**Discussion by lolgab** [com-lihaoyi/mill#2381] [Ideas]
Currently, Mill adds a:

```scala
import _root_.mill._
```
magic import.

I would love to move in a direction where the magic in the build is less and less, eventually (and hopefully) being a vanilla `build.scala` file.

So I propose to remove this magic import in Mill 0.11, so people will need to manually `import mill._` in their builds.

**Comment by lefou** [comment]
Interesting, in our documentation we often show to add `import mill._`. I wasn't even aware of the fact we don't have to import that package manually. I'm for removing it.

**Comment by lihaoyi** [comment]
I didn't realize this was done automatically. I've been doing `import mill._` at the top of my `build.sc` files for the past half decade

**Reply by lefou** [reply]
Haha, me too.

**Comment by lefou** [comment]
Is this already resolved in current main, or should we add it to `0.11.0`?

**Reply by lolgab** [reply]
No, it's not resolved yet.
I need to finish what I started here: https://github.com/com-lihaoyi/mill/pull/2418

**Reply by lefou** [reply]
@lolgab Understood. How do you estimate the needed work for it? @lihaoyi Do you think we can squeeze that in too, before `0.11.0`?

**Reply by lolgab** [reply]
@lefou I pushed some commits and merged master some minutes ago. If the CI succeedes we can merge it today :)

**Reply by lefou** [reply]
@lolgab Great, thank you!

**Comment by lefou** [comment]
This is now implemented in Mill 0.11.0.
